core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponentSpec.scala

```diff
@@ -1313,6 +1314,57 @@ class CompactEntityComponentSpec extends TestDriverComponentWithFlatSpecAndMatch
     }
   }
 
+  behavior of "listEntities"
+
+  it should "handle entities with simple attributes" in withMinimalTestDatabase { _ =>
```

Comment:
I think you are being too complicated in this test and not covering all the important cases. I think the important cases are
1. it returns more than 1 entity of the given type
2. it excludes other types
3. it excludes the same type in other workspaces

This test does only the second. I think your setup should include 2 entities of the type queried for, 1 entity of another type in the same workspace and 1 entity of the queried type but in a different workspace. Query only the main workspace as querying the other workspace does not provide additional testing value. Then verify that only the 2 expected entities are in the result. Doing stuff like 
```
val ws1EntityType = ws1Entities.head.entityType
```
and 
```
.filter(_.entityType == ws1EntityType)
```
overcomplicate the test.